XpringKit/XRP/Utils.swift

```diff
@@ -64,4 +66,75 @@ public enum Utils {
   public static func toTransactionHash(transactionBlobHex: Hex) -> Hex? {
     return javaScriptUtils.toTransactionHash(transactionBlobHex: transactionBlobHex)
   }
+
+  /// Convert from units in drops to XRP.
+  ///
+  /// - Parameter drops: An amount of XRP expressed in units of drops.
+  /// - Returns: A String representing the drops amount in units of XRP.
+  /// - Throws: An XRPLedgerError if drops is in an invalid format.
+  public static func dropsToXrp(_ drops: String) throws -> String {
+    // TODO: nil preconditions check?
+
+    let dropsRegex: String = "^-?[0-9]*['.']?[0-9]*$"
+    if drops.range(of: dropsRegex, options: .regularExpression) == nil {
+      throw XRPLedgerError.invalidDropsValue(
+        "dropsToXrp: invalid value \(drops), should be a number matching \(dropsRegex)."
+      )
+    } else if drops == "." {
+      throw XRPLedgerError.invalidDropsValue("dropsToXrp: invalid value \(drops), should be a string-encoded number.")
+    }
+
+    // check for non-zero fractional amount in drops value
+    if drops.contains(".") {
+      let dropsComponents: [Substring] = drops.split(separator: ".")
+      if dropsComponents.count > 2 {
+        throw XRPLedgerError.invalidDropsValue("dropsToXrp: invalid value, \(drops) has too many decimal points.")
+      }
+      if !drops.hasSuffix(".") {
+        let dropsFractionalAmount = NSDecimalNumber(string: String(dropsComponents[dropsComponents.count - 1]))
+        if dropsFractionalAmount != NSDecimalNumber.zero {
+          throw XRPLedgerError.invalidDropsValue("dropsToXrp: value \(drops) must be a whole number.")
+        }
+      }
+    }
+    return NSDecimalNumber(string: drops).dividing(by: NSDecimalNumber(1_000_000.0))
+      .description(withLocale: Locale(identifier: "en_US"))
+  }
+
+  /// Convert from units in XRP to drops.
+  ///
+  /// - Parameter xrp: An amount of XRP expressed in units of XRP.
+  /// - Returns: A String representing an amount of XRP expressed in units of drops.
+  /// - Throws: An XRPLedgerError if XRP is in invalid format.
+  public static func xrpToDrops(_ xrp: String) throws -> String {
+    // TODO: nil preconditions check?
```

Comment:
you can resolve this. `String` is always non-nil in swift. If your parameter was a `String?` then you'd need to `guard let` here.